src/cockpit/389-console/src/lib/security/certificateManagement.jsx

```diff
@@ -165,6 +241,98 @@ export class CertificateManagement extends React.Component {
                 });
     }
 
+    addCSR () {
+        if (this.state.csrName == "") {
+            this.props.addNotification(
+                "warning",
+                `Missing CSR Name`
+            );
+            return;
+        } else if (this.state.csrSubjectCommonName == "") {
+            this.props.addNotification(
+                "warning",
+                `Missing CSR Subject Common Name`
+            );
+            return;
+        }
+
+        this.setState({
+            modalSpinning: true,
+            loading: true,
+        });
+        const cmd = [
+            "dsconf", "-j", "ldapi://%2fvar%2frun%2fslapd-" + this.props.serverId + ".socket",
+            "security", "csr", "req", "--name=" + this.state.csrName, "--subject=" + this.state.csrSubject
+        ];
+        log_cmd("addCSR", "Creating CSR", cmd);
+        cockpit
+                .spawn(cmd, { superuser: true, err: "message" })
+                .done(() => {
+                    this.reloadCSRs();
+                    this.setState({
+                        showAddCSRModal: false,
+                        csrSubject: '',
+                        csrName: '',
+                        modalSpinning: false,
+                    });
+                    this.props.addNotification(
+                        "success",
+                        `Successfully created CSR`
+                    );
+                })
+                .fail(err => {
+                    const errMsg = JSON.parse(err);
+                    if (errMsg.desc.includes('certutil -s: improperly formatted name:')) {
+                        this.props.addNotification(
+                            "error",
+                            `Error Incorrect County Code`
```

Comment:
The message is incorrect because the wrong format is not always about the Country Code:

![image](https://user-images.githubusercontent.com/5634892/212766023-192999f3-9229-45ce-9f46-72143b5e3fdb.png)

Reply:
> The message is incorrect because the wrong format is not always about the Country Code:

From a certutil perspective, the only subject field it imposes a format on is the country code. So all the other subject fields accept anything, except a few special characters I didn't test with. Good catch.